src/venv_manager.rs

```diff
@@ -80,14 +83,51 @@ impl VenvManager {
     }
 
     pub fn run(&self, args: &[String]) -> Result<(), Error> {
-        if !self.paths.venv.exists() {
-            return Err(Error::MissingVenv {
-                path: self.paths.venv.clone(),
+        self.expect_venv()?;
+        // Compiler assumes `rc` is never read
+        // Maybe because of the unsafe block ?
+        #[allow(unused_assignments)]
+        let mut rc = 0;
+        unsafe {
+            let arg0 = &args[0].clone();
+            let bin_path = &self.get_path_in_venv(&arg0)?;
+            let bin_path_str = bin_path.to_str().ok_or(Error::Other {
+                message: "Could not convert binary path to char*".to_string(),
+            })?;
+            let mut fixed_args: Vec<String> = args.iter().map(|x| x.to_string()).collect();
+            fixed_args[0] = bin_path_str.to_string();
+
+            let mut c_args = vec![];
+            for arg in fixed_args {
+                c_args.push(CString::new(arg.as_str()).map_err(|_| Error::NulByteFound { arg })?);
+            }
+            let mut args: Vec<*const c_char> =
+                c_args.iter().map(|x| x.as_ptr() as *const c_char).collect();
```

Comment:
You don't need the `as *const c_char` if you call as_ptr, that's already the type as_ptr returns